pages/etherpad/[[...roomId]].js

```diff
@@ -208,11 +236,12 @@ export default function Etherpad() {
                                     // if the room name is undefined we don't want to display it
                                     if (!name) return;
 
-                                    return <ServiceLink
+                                    return <WritePads
                                         key={writeRoomId}
-                                        name={_.get(matrix.rooms.get(writeRoomId), 'name')}
+                                        name={name}
                                         href={`${etherpadPath}/${writeRoomId}`}
-                                        passwordProtected={serverPads[matrix.roomContents.get(writeRoomId)?.body.substring(matrix.roomContents.get(writeRoomId)?.body.lastIndexOf('/') + 1)]?.visibility === 'private'}
+                                        serverPads={serverPads}
```

Comment:
This also looks a bit weird.
So we're inside a `.map()` here, where we cycle through _each_ pad that we want to show.

And now -- for some reason -- that single pad needs to have **all the server pads** data structure? That's usually a hint that something can be improved...